src/core/storage/types.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Configuration options for the storage client
+ */
+export interface StorageConfig {
+  /** Private key for w3up-client authentication */
+  privateKey: string | undefined;
+  /** Delegation for storage access */
+  delegation: string | undefined;
+  /** Optional gateway URL for file retrieval */
+  gatewayUrl?: string;
```

Comment:
Consider requiring rich types for these configs `Signer`, `Delegation` and `URL` to push the parsing back into initialization functions (`loadConfig`) so the errors are surfaced at startup not runtime.

Obviously when providing a delegation per request, that has to be parsed in the handler at request time...